src/PrestaShopBundle/DependencyInjection/Compiler/GridDefinitionServiceIdsCollectorPass.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\GridDefinitionFactoryInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Collects grid definition service ids.
+ */
+final class GridDefinitionServiceIdsCollectorPass implements CompilerPassInterface
+{
+    const GRID_DEFINITION_SERVICE_STARTS_WITH = 'prestashop.core.grid.definition';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!in_array($container->getParameter('kernel.environment'), ['dev', 'test'])) {
+            return;
+        }
+
+        $serviceDefinitions = $container->getDefinitions();
+        $gridServiceIds = [];
+
+        foreach ($serviceDefinitions as $serviceId => $serviceDefinition) {
+            if ($serviceDefinition->isAbstract() || $serviceDefinition->isPrivate()) {
+                continue;
+            }
+
+            if ($this->isGridDefinitionService($serviceId, $serviceDefinition->getClass())) {
+                $gridServiceIds[] = $serviceId;
+            }
+        }
+
+        $container->setParameter(
+            'prestashop.core.grid.definition.service_ids',
+            $gridServiceIds
+        );
+    }
+
+    /**
+     * Checks if grid definition service.
+     *
+     * @param string $serviceId
+     * @param string $serviceClass
+     *
+     * @return bool
+     */
+    private function isGridDefinitionService($serviceId, $serviceClass)
+    {
+        $doesServiceStartsWithGridDefinition = strpos($serviceId, self::GRID_DEFINITION_SERVICE_STARTS_WITH) === 0;
```

Comment:
I guess you perform this test to optimize your search? But this means if a service was not defined properly it won't be detected?
We may be wary of this for core services but what about module services? I'm wondering if we could use tags and autowiring to easily fond those interfaces

Reply:
https://symfony.com/blog/new-in-symfony-3-3-service-autoconfiguration

Reply:
yep, I 100% agree with you but here's the catch:

I need this functionality to list hooks as mentioned in the issue https://github.com/PrestaShop/PrestaShop/issues/13408. The hooks list is required todo in 1.7.6.x version which means I can't add tags to the services. And also, since its only core hooks that are needed to display this  method atleast for now looks good enough